sunpy/map/mapbase.py

```diff
@@ -243,7 +246,51 @@ def __repr__(self):
                                refpix=u.Quantity(self.reference_pixel),
                                refcoord=u.Quantity((self.reference_coordinate.data.lon,
                                                     self.reference_coordinate.data.lat)),
-                               tmf=TIME_FORMAT) + self.data.__repr__()
+                               tmf=TIME_FORMAT)
+
+    def __repr__(self):
+        return object.__repr__(self) + "\n" + self._text_summary() + "\n" + self.data.__repr__()
+
+    def _repr_html_(self):
+        buf = BytesIO()
+        fig = Figure(figsize=(4.8, 4.8), constrained_layout=True)
+        ax = fig.subplots()
+        ax.imshow(self.data, cmap='gray', origin='lower',
+                  norm=ImageNormalize(stretch=HistEqStretch(self.data)))
+        ax.set_xlabel('X pixel')
+        ax.set_ylabel('Y pixel')
+        ax.set_title('Histogram-equalized image')
+        fig.savefig(buf, format='png')
+        image_src = b64encode(buf.getvalue()).decode('utf-8')
+
+        buf = BytesIO()
+        fig = Figure(figsize=(4.8, 2.4), constrained_layout=True)
+        ax = fig.subplots()
+        ax.hist(self.data.ravel(), bins=100, histtype='stepfilled')
```

Comment:
I think setting bins=100 will fail if any of the map data contains a NaN, so this should probably be `np.linspace(np.nanmin(self.data), np.nanmax(self.data), 100)`.

Reply:
NaNs don't appear to cause any issue (at least on my version of Matplotlib).

Reply:
I added an example in the original post for a Map with NaNs.

Reply:
Ah maybe it's infs that I've had problems with in the past

Reply:
It turns out that I missed warnings due to the NaNs.  The code now better handles infinites and NaNs.